src/cript/api/api.py

```diff
@@ -402,6 +406,791 @@ def api_prefix(self):
+            """
+            url_path = f"/{parent_node0}/{parent_uuid0}"
+            payload_removes = item["payload_json_removes"]
+
+            if payload_removes != {}:
+                try:
+                    # payload_remove needs --- {"node": new_node.node}:
+                    API.add_to_dict(payload_removes, key="node", value=[parent_node0.capitalize()])
+                    # remove_response = self._capsule_request(url_path=url_path, method="PATCH", data=json.dumps(payload_removes))
+
+                    self._capsule_request(url_path=url_path, method="PATCH", data=json.dumps(payload_removes))
+
+                except:
+                    print("could not remove")
+                    pass
+
+    ###########################################################################################
+    # """
+    # ok the way this would work is if I would take in two objects ,
+    #  - first i map put the paths of the first object, all uuids
+    #  - then I iterate of all uuids in the second object (I'm walking here)
+    #  - then with each uuid I encounter in the second walker node
+    #  - i find up in the lookup table and do a compare on that as the root , right ?
+    #     - basically that path would be the spot to get to in the object
+    #     - and i would record the patches and removes on each node
+    #     - I guess I could log it all to a list and maybe eventually do a group by on it
+
+    #  - if I don't find the uuid in the lookup table then it was added
+    #  - also I would need to make sure theres "visited" aspect in the iterator
+    #  - also since this object is just a map , it doesn't matter the order
+    #  - we will still probably get it in DFS because thats how iterator is
+
+    # """
+    def save_node(self, new_node: PrimaryBaseNode, link_existing=True):
+        # ================trying
+        list_of_patches_and_removes = []
+
+        node_type = new_node.node_type.lower()
+
+        try:
+            get_url = f"/{node_type}/{new_node.uuid}"
+
+            try:
+                original = self._capsule_request(url_path=get_url, method="GET").json()
+                original = original["data"][0]
+
+                # print("\noriginal 1: ", original)
+
+            except Exception as e:  # Exception If We Could Not Load By uuid
+                original = None
+                # raise ValueError(f"No data available in response {response}")
+
+                if node_type == "project":  # only project node is name unique
+                    klass = globals().get(node_type.capitalize(), None)
+
+                    paginator = self.search(node_type=klass, search_mode=SearchModes.EXACT_NAME, value_to_search=new_node.name)  # value["name"])
+
+                    klass_json = next(paginator)
+
+                    if (original is None) and klass_json:
+                        raise ValueError("this name already exists stored under a different uuid")
```

Comment:
we need better errors.
Please make custom Exception ideally with recommendations for users and throw these.